core/src/test/scala/integration/kafka/admin/AdminFenceProducersIntegrationTest.scala

```diff
@@ -107,6 +108,37 @@ class AdminFenceProducersIntegrationTest extends IntegrationTestHarness {
     assertThrows(classOf[ProducerFencedException], () => producer.commitTransaction())
   }
 
+  @ParameterizedTest
+  @ValueSource(strings = Array("zk", "kraft"))
+  def testFenceProducerTimeoutMs(quorum: String): Unit = {
+    producer.initTransactions()
+    producer.beginTransaction()
+    producer.send(record).get()
+    producer.commitTransaction()
+    def timeoutOption = new FenceProducersOptions()
+    try {
+      adminClient.fenceProducers(Collections.singletonList(txnId), timeoutOption.timeoutMs(0)).all().get()
```

Comment:
this will be flaky. It can be completed if the request is executed immediately. Please create a `Admin` with invalid bootstrap and then pass a request having timeout=0. In that scenario, the request should return timeout exception. see https://github.com/apache/kafka/blob/0cbc5e083a936025a85f127102dc1032f6cf4fd9/core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala#L159 for example

Reply:
i'm sorry that i forget that